ranch/storage.go

```diff
@@ -108,7 +108,7 @@ func (s *Storage) GetResource(name string) (*crds.ResourceObject, error) {
 		return nil, fmt.Errorf("failed to get resource %s: %v", name, err)
 	}
 	if o.Status.UserData == nil {
-		o.Status.UserData = &common.UserData{}
+		o.Status.UserData = map[string]string{}
```

Comment:
nitful suggestion: I think we have a `common.UserDataMap` type that you could use here (and elsewhere) instead of `map[string]string`.

Reply:
the `common.UserDataMap` is aliasing a `map[string]string` without adding any methods which IMHO makes the code that uses it harder to understand without adding any benefit that I can see. I'd rather remove `common.UserDataMap` in a follow-up.